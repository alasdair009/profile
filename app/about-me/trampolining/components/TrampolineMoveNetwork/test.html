<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
