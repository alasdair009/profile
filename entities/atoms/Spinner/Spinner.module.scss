@use "../../../app/styles/animation";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

@keyframes rotate {
  to {
    transform: rotate(360deg);
  }
}

.root {
  content: "";
  border-radius: 50%;
  display: block;
  height: rem.toRem(map.get(dimensions.$sizes, s48));
  width: rem.toRem(map.get(dimensions.$sizes, s48));
}

.spinnerImage {
  animation: rotate animation.animationDuration(1)
    map.get(animation.$curves, linear) infinite;
  height: 100%;
  object-fit: contain;
  width: 100%;
}
