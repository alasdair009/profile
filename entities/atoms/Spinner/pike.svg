<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="183px" height="258px" style="shape-rendering:geometricPrecision; text-rendering:geometricPrecision; image-rendering:optimizeQuality; fill-rule:evenodd; clip-rule:evenodd" xmlns:xlink="http://www.w3.org/1999/xlink">
<g><path style="opacity:0.942" fill="#fff" d="M 164.5,-0.5 C 165.5,-0.5 166.5,-0.5 167.5,-0.5C 170.981,1.33775 174.647,3.00442 178.5,4.5C 179.663,9.80234 179.829,15.1357 179,20.5C 178.626,21.4162 178.126,22.2496 177.5,23C 178.204,28.5096 179.204,34.0096 180.5,39.5C 180.471,41.0293 179.804,42.196 178.5,43C 179.977,44.8121 181.31,46.6455 182.5,48.5C 182.5,50.1667 182.5,51.8333 182.5,53.5C 178.974,55.0104 176.14,57.3437 174,60.5C 171.639,65.9174 170.306,71.5841 170,77.5C 168.92,79.9962 167.586,82.3295 166,84.5C 165.984,91.0613 164.484,97.228 161.5,103C 158.491,102.84 156.325,104.007 155,106.5C 138.143,153.564 118.476,199.564 96,244.5C 91.2888,251.066 85.1222,255.399 77.5,257.5C 73.5,257.5 69.5,257.5 65.5,257.5C 59.0001,255.254 52.6668,252.421 46.5,249C 33.4847,235.993 21.9847,221.826 12,206.5C 7.50878,197.195 3.34212,187.861 -0.5,178.5C -0.5,169.5 -0.5,160.5 -0.5,151.5C 1.33483,148.158 2.5015,144.491 3,140.5C 4.35613,139.62 5.85613,139.286 7.5,139.5C 7.33788,137.473 7.50454,135.473 8,133.5C 12.2237,127.719 15.3904,121.386 17.5,114.5C 14.9469,109.771 11.7803,105.438 8,101.5C 4.48667,94.0387 3.82,86.372 6,78.5C 13.9482,68.1107 24.4482,63.6107 37.5,65C 45.9522,69.7852 52.7855,76.2852 58,84.5C 59.3706,91.7761 60.204,99.1094 60.5,106.5C 63.6907,107.092 66.6907,108.259 69.5,110C 73.4862,110.499 77.4862,110.666 81.5,110.5C 81.2308,108.571 81.7308,106.905 83,105.5C 83.5103,108.005 85.0103,109.505 87.5,110C 92.8333,110.667 98.1667,110.667 103.5,110C 107.361,108.774 109.694,106.274 110.5,102.5C 111.822,102.67 112.989,102.337 114,101.5C 120.667,90.8333 127.333,80.1667 134,69.5C 140.137,53.4207 146.471,37.4207 153,21.5C 151.299,11.2335 155.133,3.90021 164.5,-0.5 Z M 58.5,139.5 C 67.8856,140.892 76.8856,142.558 85.5,144.5C 83.0979,152.312 78.9312,158.978 73,164.5C 70.6729,169.825 67.8395,174.825 64.5,179.5C 63.914,179.291 63.414,178.957 63,178.5C 62.6667,171.5 62.3333,164.5 62,157.5C 60.5121,151.551 59.3454,145.551 58.5,139.5 Z"/></g>
</svg>
