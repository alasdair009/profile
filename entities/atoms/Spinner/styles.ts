"use client";
import styled, { keyframes } from "styled-components";
import { animationDurationCSS, curves, sizes } from "@/entities";
import Image from "next/image";

const rotate = keyframes`
  to {
    transform: rotate(360deg);
  }
`;
export const Root = styled.div`
  content: "";
  border-radius: 50%;
  display: block;
  height: ${sizes.s48.rem};
  width: ${sizes.s48.rem};
`;

export const SpinnerImage = styled(Image)`
  animation: ${rotate} ${animationDurationCSS(1)} ${curves.linear} infinite;
  height: 100%;
  object-fit: contain;
  width: 100%;
`;
