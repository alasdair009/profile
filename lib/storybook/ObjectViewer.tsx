import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
