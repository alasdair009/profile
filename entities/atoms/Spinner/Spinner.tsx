import { Root, SpinnerImage } from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import pikeSvg from "./pike.svg";
import { Heading } from "@/entities";

export function Spinner({ ...rest }: HTMLAttributes<HTMLDivElement>) {
  return (
    <Root {...rest}>
      <SpinnerImage src={pikeSvg} alt="A gymnast in a pike shape" />
    </Root>
  );
}
