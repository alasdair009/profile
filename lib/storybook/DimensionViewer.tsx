import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
