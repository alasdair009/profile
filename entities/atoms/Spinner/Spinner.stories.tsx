import { Spinner } from "./Spinner";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";

const meta: Meta<typeof Spinner> = {
  component: Spinner,
};
export default meta;

export const Default: StoryObj<typeof Spinner> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const spinnerElement = canvas.getByTestId(Spinner.name);

    await expect(spinnerElement).toBeInTheDocument();
  },
};
