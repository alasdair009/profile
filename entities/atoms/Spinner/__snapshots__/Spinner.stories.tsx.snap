// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Spinner Default play-test 1`] = `
<div data-testid="Spinner"
     class="sc-drnuxz guAkRS"
>
  <img alt="A gymnast in a pike shape"
       loading="lazy"
       width="183"
       height="258"
       decoding="async"
       data-nimg="1"
       class="sc-hDcvty gQRnHp"
       srcset="/static/media/entities/atoms/Spinner/pike.svg?w=256&amp;q=75 1x, /static/media/entities/atoms/Spinner/pike.svg?w=384&amp;q=75 2x"
       src="/static/media/entities/atoms/Spinner/pike.svg?w=384&amp;q=75"
       style="color: transparent;"
  >
</div>
`;
